### PR TITLE
Add support of Linux and macOS

### PR DESCRIPTION
**Pull request by dermeister0**
## Description

- Check build success in psake.ps1 instead of psake.cmd.
- Add shebang to script.
- Add +x permission to psake.ps1 file.

## Related Issue

#262 

## Motivation and Context

It allows to use psake.ps1 to run psake tasks on Linux and macOS. Also psake.ps1 returns correct exit code now.

## How Has This Been Tested?

I created tasks that may produce errors:

```powershell
Task test `
{
    Hi-There
}
```

```sh
echo 'Task test { Hi-There }' > Test1.ps1
```

```powershell
Task test `
    -requiredVariables @('SomeValue') `
{
    "Test $SomeValue"
}
```

```sh
echo "Task test -requiredVariables @('SomeValue') { \"Test \$SomeValue\" }" > Test2.ps1
```

```powershell
Task test `
{
    $PSVersionTable
}
```

```sh
echo 'Task test { $PSVersionTable }' > Test3.ps1
```

Tested them on Windows 10 (PowerShell 5.1) and Ubuntu 16.04 (PowerShell Core 6.1).

```sh
psake.ps1 Test1.ps1 ; echo $?
psake.ps1 Test2.ps1 ; echo $?
psake.ps1 Test2.ps1 -parameters '@{SomeValue=123}' ; echo $?
psake.ps1 Test3.ps1 ; echo $?
```

Made sure correct exit code is returned.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by dermeister0** [comment]
> Also, did you run the Git command to mark psake.ps1 as executable?

I ran it but lost this change somehow. Fixed now. Also I had to change `autocrlf` settings for psake.ps1, because `.gitattributes` uses `crlf` line endings for all files.

```
der@build:~$ git clone https://github.com/Saritasa/psake psake-fix-test
Cloning into 'psake-fix-test'...
remote: Enumerating objects: 14, done.
remote: Counting objects: 100% (14/14), done.
remote: Compressing objects: 100% (10/10), done.
remote: Total 2093 (delta 5), reused 10 (delta 3), pack-reused 2079
Receiving objects: 100% (2093/2093), 3.16 MiB | 472.00 KiB/s, done.
Resolving deltas: 100% (1144/1144), done.
der@build:~$ cd psake-fix-test/
der@build:~/psake-fix-test$ ls -l src/psake.ps1
-rw-rw-r-- 1 der der 2211 Oct  9 02:36 src/psake.ps1
der@build:~/psake-fix-test$ git checkout fix/add-support-of-linux-and-macos
Branch 'fix/add-support-of-linux-and-macos' set up to track remote branch 'fix/add-support-of-linux-and-macos' from 'origin'.
Switched to a new branch 'fix/add-support-of-linux-and-macos'
der@build:~/psake-fix-test$ ls -l src/psake.ps1
-rwxrwxr-x 1 der der 2207 Oct  9 02:36 src/psake.ps1
der@build:~/psake-fix-test$ src/psake.ps1
psake version 4.7.4
Copyright (c) 2010-2017 James Kovacs & Contributors

Error: 10/9/18 2:36:40 AM:
At /home/der/psake-fix-test/src/public/Assert.ps1:69 char:9 +         throw ('Assert: {0}' -f $failureMessage) +         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ [<<==>>] Exception: Assert: Could not find the build file psakefile.ps1.
der@build:~/psake-fix-test$ src/psake.ps1 --help

NAME
    Invoke-psake

SYNOPSIS
    Runs a psake build script.
```

File: .gitattributes [hunk 1]
```diff
@@ -1 +1,2 @@
-* -crlf
\ No newline at end of file
+* -crlf
+src/psake.ps1 text
```
Comment (rkeithhill) [inline .gitattributes]: Maybe this project should use a more typical .gitattributes e.g.:
```
CHANGELOG.md merge=union
* text=auto
```

Comment (dermeister0) [reply .gitattributes]: Reported a new issue: #266

**Comment by dermeister0** [comment]
```
der@build:~$ psake-fix-test/src/psake.ps1 Test1.ps1 test ; echo $?
psake version 4.7.4
Copyright (c) 2010-2017 James Kovacs & Contributors

Executing test
Error: 10/9/18 3:13:48 AM:
At /home/der/Test1.ps1:1 char:13 + Task test { Hi-There } +             ~~~~~~~~ [<<==>>] Exception: The term 'Hi-There' is not recognized as the name of a cmdlet, function, script file, or operable program.Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
1
```

```
der@build:~$ psake-fix-test/src/psake.ps1 Test2.ps1 test ; echo $?
psake version 4.7.4
Copyright (c) 2010-2017 James Kovacs & Contributors

Executing test
Error: 10/9/18 3:14:32 AM:
At /home/der/psake-fix-test/src/public/Assert.ps1:69 char:9 +         throw ('Assert: {0}' -f $failureMessage) +         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ [<<==>>] Exception: Assert: Variable SomeValue must be set to run task test.
1
```

```
der@build:~$ psake-fix-test/src/psake.ps1 Test3.ps1 test ; echo $?
psake version 4.7.4
Copyright (c) 2010-2017 James Kovacs & Contributors

Executing test

Name                           Value
----                           -----
PSVersion                      6.1.0
PSEdition                      Core
GitCommitId                    6.1.0
OS                             Linux 4.15.0-36-generic #39-Ubuntu SMP Mon Sep 24 16:19:09 UTC 2018
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

psake succeeded executing Test3.ps1

----------------------------------------------------------------------
Build Time Report
----------------------------------------------------------------------
Name   Duration
----   --------
test   00:00:00.045
Total: 00:00:00.101


0
```

**Comment by dermeister0** [comment]
Got a bug with Hashtable parsing.

```
der@build:~$ psake-fix-test/src/psake.ps1 Test2.ps1 test -parameters '@{SomeValue=123}' ; echo $?
/home/der/psake-fix-test/src/psake.ps1 : Cannot process argument transformation on parameter 'parameters'. Cannot convert the "@{SomeValue=123}" value of type "System.String" to type "System.Collections.Hashtable".
+ CategoryInfo          : InvalidData: (:) [psake.ps1], ParentContainsErrorRecordException
+ FullyQualifiedErrorId : ParameterArgumentTransformationError,psake.ps1

1
```

It works with a wrapper.

```
der@build:~$ psake Test2.ps1 test -parameters '@{SomeValue=123}' ; echo $?         
psake version 4.7.4                                                                
Copyright (c) 2010-2017 James Kovacs & Contributors                                
                                                                                   
Executing test                                                                     
Test 123                                                                           
                                                                                   
psake succeeded executing Test2.ps1                                                
                                                                                   
----------------------------------------------------------------------             
Build Time Report                                                                  
----------------------------------------------------------------------             
Name   Duration                                                                    
----   --------                                                                    
test   00:00:00.057                                                                
Total: 00:00:00.123                                                                
0                                                                                  
```

**Comment by dermeister0** [comment]
The approach does not allow to run psake scripts from shells different from `pwsh`. I created wrapper-based implementation instead in another pull request: 267

**Pull request closed without merging**